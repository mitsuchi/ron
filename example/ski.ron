# 演算子を宣言
op 50 : _ -> _
op 50 : _ => _
op 100 : _ _

syntax {
    constants ::= I | K | S
}

# 規則1
I x     -> x
K x y   -> x
S x y z -> x z (y z)

# 規則2
x y -> x' y { x -> x' }
x y -> x y' { y -> y' }

# 規則3
x => y { x -> y }
x => y { x => z; z => y }

# 問い合わせ
main {
    S K S K => x
}