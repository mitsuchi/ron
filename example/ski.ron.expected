x = K K (S K)